parsers/tidb/tidb.go

```diff
@@ -0,0 +1,448 @@
+/**
+* @Author: Fan()
+* @Date: 2022/10/18
+ */
+
+package tidb
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/parsers"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/execdetails"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/plancodec"
+	"github.com/sirupsen/logrus"
+)
+
+const timeFormat = "2006-01-02 15:04:05.999999999"
+
+type Options struct {
+	NumParsers int `hidden:"true" description:"number of TIDB parsers to spin up" yaml:"-"`
+}
+
+// Parser is used to read slow log data.
+type Parser struct {
+	outputCols            []string
+	initialized           bool
+	columnValueFactoryMap map[string]slowQueryColumnValueFactory
+	SampleRate            int
```

Comment:
```
// Event is a single log event
type Event struct {
	// Timestamp is the time of the event (may be different from current time)
	Timestamp time.Time
	// SampleRate is the rate at which this event is sampled. If it is positive,
	// the event should be sent with that sample rate.  If it is -1 the event
	// should be dropped instead of getting sent. Zero value should be treated as
	// unset and the event sent with a sample rate of 1
	SampleRate int
	// Data is a map[string]interface{} containing key/value pairs for all the
	// metrics to submit in this event
	Data map[string]interface{}
}
```
Event.SampleRate is int. Of course, I can change this piece as well